### PR TITLE
Adding validator for payment methods

### PR DESCRIPTION
**Pull request by mamazu**

### CODE REVIEW

File: spec/Validator/Cart/PaymentMethodAvailableValidatorSpec.php [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        $orderRepository->findOneBy(['tokenValue' => 'NON_EXISTING', 'state' => 'cart'])->willReturn($cart);
+        $cart->getPayments()->willReturn(new ArrayCollection([$payment->getWrappedObject()]));
+
+        $paymentMethodsResolver->getSupportedMethods($payment)->shouldBeCalled()->willReturn([$paymentMethod]);
+        $paymentMethod->getCode()->willReturn('paypal');
+
+        $context->buildViolation('sylius.shop_api.checkout.payment_method_not_available')->willReturn(
+            $violationBuilder
+        )
+        ;
```
Comment (lchrusciel) [inline spec/Validator/Cart/PaymentMethodAvailableValidatorSpec.php:68]:
```suggestion
        );
```

File: src/Validator/Cart/PaymentMethodAvailableValidator.php [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        OrderRepositoryInterface $orderRepository,
+        PaymentMethodsResolverInterface $paymentMethodsResolver
+    ) {
+        $this->orderRepository = $orderRepository;
+        $this->paymentMethodsResolver = $paymentMethodsResolver;
+    }
+
+    public function validate($value, Constraint $constraint): void
+    {
+        /** @var ChoosePaymentMethodRequest $value */
```
Comment (lchrusciel) [inline src/Validator/Cart/PaymentMethodAvailableValidator.php:35]: What do you think about adding an assertion here as well?

File: src/Validator/Constraints/PaymentMethodExists.php [hunk 3]
```diff
@@ -0,0 +1,19 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sylius\ShopApiPlugin\Validator\Constraints;
+
+use Sylius\ShopApiPlugin\Validator\Cart\PaymentMethodExistsValidator;
+use Symfony\Component\Validator\Constraint;
+
+final class PaymentMethodExists extends Constraint
```
Comment (lchrusciel) [inline src/Validator/Constraints/PaymentMethodExists.php:10]: Shouldn't it be only property constraint?

File: src/Validator/Cart/PaymentMethodExistsValidator.php [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+    public function __construct(PaymentMethodRepositoryInterface $paymentMethodRepository)
+    {
+        $this->paymentMethodRepository = $paymentMethodRepository;
+    }
+
+    /** {@inheritdoc} */
+    public function validate($value, Constraint $constraint): void
+    {
+        Assert::isInstanceOf($constraint, PaymentMethodExists::class);
```
Comment (lchrusciel) [inline src/Validator/Cart/PaymentMethodExistsValidator.php:26]: I'm not sure about having two different behaviors in two validators in the same PR. Can we either ensure constraint or skip it in both?

Comment (mamazu) [reply src/Validator/Cart/PaymentMethodExistsValidator.php:26]: So you mean if a payment method does not exist it is also not valid for the order? So we should validate that the order can have this payment method?

File: src/Validator/Constraints/PaymentMethodExists.php (hunk 3 above)
Comment (mamazu) [reply src/Validator/Constraints/PaymentMethodExists.php:10]: Yes, by default it is but I added the explicit `getTargets` method to the constraint now.

File: src/Validator/Cart/PaymentMethodAvailableValidator.php (hunk 2 above)
Comment (mamazu) [reply src/Validator/Cart/PaymentMethodAvailableValidator.php:37]: Yes, forgot that.

File: src/Validator/Cart/PaymentMethodExistsValidator.php (hunk 4 above)
Comment (lchrusciel) [reply src/Validator/Cart/PaymentMethodExistsValidator.php:26]: No, just having here `Assert::isInstanceOf($constraint, PaymentMethodExists::class);` and there is nothing alike in the second validator

Comment (mamazu) [reply src/Validator/Cart/PaymentMethodExistsValidator.php:26]: Done that. Yeah, forgot that.

**Comment by lchrusciel** [comment]
Thanks, @mamazu! :tada:

**Pull request merged**